docs/tutorials/complete_applications/complete_application/step_05/step_05.md

```diff
@@ -0,0 +1,124 @@
+|>
+
+
+<center><|Compare primarys|button|on_action=compare|></center>
+```
+
+The Markdown section defines the layout and components of the Performance Page. It includes the following elements:
+
+- **Conditional Rendering**: `<|part|render={len(comparison_scenario)>0}|...|>`
+
+The part component ensures that certain elements are only rendered when there is data available in the *comparison_scenario*.
+
+- **Table**: `<|{comparison_scenario}|table|>`
+
+A table that displays the comparison data from the *comparison_scenario* DataFrame. The table is expandable if the number of rows exceeds the available space.
+
+- **Metric Selector**: `<|{selected_metric}|selector|lov={metric_selector}|dropdown|>`
+
+A dropdown selector that allows users to choose between two metrics: Root Mean Squared Error (RMSE) and Mean Absolute Error (MAE).
+
+- **Bar Charts**: `<|{comparison_scenario}|chart|type=bar|x=Scenario Name|y[1]=RMSE baseline|y[2]=RMSE ML|...|>`
+
+Two bar charts that compare the selected metric (RMSE or MAE) between baseline and machine learning predictions. The charts display the performance metrics for each scenario.
+
+- **Compare Button**: `<|Compare primarys|button|on_action=compare|>`
+
+A button that triggers the "compare" action when clicked. It is used to initiate the comparison process.
+
+## Python Code (pages/performance/performance.py)
+
+The Python code initializes and manages the state of the Performance Page.
+
+```python
+from taipy.gui import Markdown
+
+import pandas as pd
+import taipy as tp
+
+# Initial dataset for comparison
+comparison_scenario = pd.DataFrame(columns=["Scenario Name",
+                                            "RMSE baseline",
+                                            "MAE baseline",
+                                            "RMSE ML",
+                                            "MAE ML"])
+
+
+# Selector for metrics
+metric_selector = ["RMSE", "MAE"]
+selected_metric = metric_selector[0]
+
+
+def compare(state):
+    print("Comparing...")
+
+    # Initialize lists for comparison
+    scenario_data = []
+
+    # Go through all the primary scenarios
+    all_scenarios = sorted(tp.get_primary_scenarios(), key=lambda x: x.creation_date.timestamp())
+
+    for scenario in all_scenarios:
+        rmse_baseline, mae_baseline = scenario.metrics_baseline.read()
+        rmse_ml, mae_ml = scenario.metrics_ml.read()
+
+
+        # Store scenario data in a dictionary
+        scenario_data.append({
+            "Scenario Name": scenario.name,
+            "RMSE baseline": rmse_baseline,
+            "MAE baseline": mae_baseline,
+            "RMSE ML": rmse_ml,
+            "MAE ML": mae_ml
+        })
+
+    # Create a DataFrame from the scenario_data list
+    state.comparison_scenario = pd.DataFrame(scenario_data)
+
+performance = Markdown("pages/performance/performance.md")
+```
+
+It includes the following components:
```

Comment:
It is not defined here. At this point, I would just remove the line that does not bring any information.

Reply:
Which line?

Reply:
105. The word "it" does not refer to anything.